include/mbedtls/mbedtls_config.h

```diff
@@ -3680,8 +3680,56 @@
 
 /* Platform options */
 //#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h> /**< Header to include if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS is defined. Don't define if no header is needed. */
-//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc /**< Default allocator to use, can be undefined */
-//#define MBEDTLS_PLATFORM_STD_FREE            free /**< Default free to use, can be undefined */
+
+/* An overview of how the value of mbedtls_calloc is determined:
+ *
+ * if !MBEDTLS_PLATFORM_MEMORY
+ *     mbedtls_calloc = calloc
+ * if MBEDTLS_PLATFORM_MEMORY
+ *     if (MBEDTLS_PLATFORM_CALLOC_MACRO && MBEDTLS_PLATFORM_FREE_MACRO):
+ *         mbedtls_calloc = MBEDTLS_PLATFORM_CALLOC_MACRO
+ *     if !(MBEDTLS_PLATFORM_CALLOC_MACRO && MBEDTLS_PLATFORM_FREE_MACRO):
+ *         Dynamic setup via mbedtls_platform_set_calloc_free is now possible with a default value MBEDTLS_PLATFORM_STD_CALLOC.
+ *         How is MBEDTLS_PLATFORM_STD_CALLOC handled?
+ *         if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS:
+ *             MBEDTLS_PLATFORM_STD_CALLOC is not set to anything;
+ *             MBEDTLS_PLATFORM_STD_MEM_HDR can be included if present;
+ *         if !MBEDTLS_PLATFORM_NO_STD_FUNCTIONS:
+ *             if MBEDTLS_PLATFORM_STD_CALLOC is present:
+ *                 User-defined MBEDTLS_PLATFORM_STD_CALLOC is respected;
+ *             if !MBEDTLS_PLATFORM_STD_CALLOC:
+ *                 MBEDTLS_PLATFORM_STD_CALLOC = calloc
+ *
+ *         At this point the presence of MBEDTLS_PLATFORM_STD_CALLOC is checked.
+ *         if !MBEDTLS_PLATFORM_STD_CALLOC
+ *             MBEDTLS_PLATFORM_STD_CALLOC = uninitialized_calloc
```

Comment:
We should state that this always fails (returning a null pointer), and the corresponding “uninitialized free” does nothing.